### PR TITLE
ref(proj-config): Introduce new tasks [INGEST-1362 INGEST-1324]

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Deleting the debounce key at the end of the task in the `finally` block ensures the key is always deleted when the whole task runs. `acks_late` option will be added in a follow-up PR.

### CODE REVIEW

File: src/sentry/tasks/relay.py [hunk 1]
```diff
@@ -47,71 +47,55 @@ def update_config_cache(
     sentry_sdk.set_tag("update_reason", update_reason)
     sentry_sdk.set_tag("generate", generate)
 
-    if not should_update_cache(
-        organization_id=organization_id, project_id=project_id, public_key=public_key
-    ):
```
Comment (iker-barriocanal) [inline src/sentry/tasks/relay.py:52]: Setting the debounce key before enqueuing the task is not transactional and leaves a race condition in which multiple tasks could be scheduled. For now, we accept these duplicates and don't make further attempts to deduplicate them.

**Review by jjbayer — COMMENTED** [review]
This looks good to me, but we should not deploy this without also implementing the strategy for invalidation tasks. The code in this PR is similar to what we used for our [latest debouncing experiment](https://github.com/getsentry/sentry/pull/35107), which we [reverted](https://github.com/getsentry/sentry/pull/35186).

**Review by untitaker — COMMENTED** [review]
My understanding is that we need two tasks: one for invalidation where the debouncing happens at the beginning, and another where it happens at the end. So I don't think this change can be applied to master as it also changes the invalidation tasks.

**Comment by flub** [comment]
we'll treat this PR as a feature branch and add the invalidation task etc here

**Review by flub — APPROVED** [review]

File: src/sentry/api/endpoints/relay/project_configs.py [hunk 2]
```diff
@@ -72,6 +67,47 @@ def _post(self, request: Request):
         else:
             return Response("Unsupported version, we only support versions 1 to 3.", 400)
 
+    def _should_use_v3(self, request):
+        version = request.GET.get("version")
```
Comment (flub) [inline src/sentry/api/endpoints/relay/project_configs.py:71]: On line 49 this code has a `or "1"` bit presumably to support very old requests.  I guess this is a bug waiting to be triggered.  We should probably either pass version in as an argument here or duplicate the line entirely.  (Personally I'd go with passing it as an argument I think)

**Review by flub — APPROVED** [review]

**Pull request merged**